code/graphics/software/FontManager.cpp

```diff
@@ -310,8 +310,117 @@ namespace font
 		}
 	}
 
+	// This function will extract the font family name (Name ID 1) from TrueType font data.
+	// It handles UCS-2 (UTF-16BE) to UTF-8 conversion.
+	static SCP_string extractFamilyNameFromTTF(const ubyte* data, int size)
+	{
+		// TTF/OTF fonts start the table directory at byte 12.
+		int table_offset = 12;
+		
+		if (size < table_offset) {
+			return "Unknown";
+		}
+
+		// Offset to the start of the table directory (after SFNT header)
+		const uint8_t* tableDir = data + table_offset;
+
+		// Read numTables (ushort at offset 4 in SFNT header)
+		// The SFNT header bytes are big-endian
+		uint16_t numTables = (static_cast<uint16_t>(data[4]) << 8) | static_cast<uint16_t>(data[5]);
+
+		const uint8_t* nameTable = nullptr;
+
+		// Iterate through table directory entries to find the 'name' table
+		for (int i = 0; i < numTables; ++i) {
+			const uint8_t* entry = tableDir + i * 16; // Each entry is 16 bytes
```

Comment:
I think we should check size is not exceeded by the data about to be read here and in a few other places to protect against corrupted/junk data and crashes.  To clarify check `size < tableDir + ( i + 1) * 16` and return

Reply:
Yeah that's good. I added this, the other tests you recommended, and maybe one or two others I noticed. I also retooled it into a try/catch just in case there's something else we missed so that it's a lot safer.